template/src/rtm-events-api/Events.ts

```diff
@@ -103,10 +103,15 @@ class Events {
     rtmPayload: RTMAttributePayload,
     toUid?: ReceiverUid,
   ) => {
-    const to = typeof toUid == 'string' ? parseInt(toUid) : toUid;
-    const rtmEngine: RtmEngine = RTMEngine.getInstance().engine;
+    const to = typeof toUid === 'string' ? parseInt(toUid, 10) : toUid;
 
     const text = JSON.stringify(rtmPayload);
+
+    if (!RTMEngine.getInstance().isEngineReady) {
+      throw new Error('RTM Engine is not ready. Call setLocalUID() first.');
```

Comment:
While throwing the error, the event will get lost, right? Can we log data to find out what are events firing event before isEngineReady

Reply:
I have made changes, now we dont throw error just print them in main send method, keeping the existing flow and printing the error .